### PR TITLE
[CS-1518] To mark messages as persistent for an exchange group

### PR DESCRIPTION
**Pull request by ergurjeet**
https://github.com/arobson/rabbot/blob/master/docs/topology.md#rabbotaddexchange-exchangename-exchangetype-options-connectionname-
Instead of setting on every exchange in a group
```
{
  "exchangeGroups": {
    "exchange.test.request.v1": {
      "exchange": {
        "persistent": true
      },
      "retryExchange": {
        "persistent": true
      },
      "rejectedExchange": {
        "persistent": true
      },
      "keys": "exchange.test"
    }
  }
}
```

Allow marking an entire group to persist messages
```
{
  "exchangeGroups": {
    "exchange.test.request.v1": {
      "persistent": true,
      "keys": "exchange.test"
    }
  }
} 
```

### CODE REVIEW

File: src/exchangeGroups.js [hunk 1]
```diff
@@ -12,15 +12,15 @@ const queueDefaults = {
...
   let returnVal = {};
   if (_.isString(param)) {
     returnVal = { name: param };
   } else if (_.isObject(param)) {
     returnVal = param;
   }
-
-  returnVal.name = returnVal.name || defaultName;
+  returnVal.name = returnVal.name || name;
+  returnVal.persistent = returnVal.persistent || persistent;
```
Comment (vtovar) [inline src/exchangeGroups.js:23]: it looks like returnVal.persistent is not being assigned so, probably it is not necessary to compare its value..

Comment (ergurjeet) [reply src/exchangeGroups.js:23]: That is to allow if you only want to persist messages in 1 exchange, say only in `rejectedExchange`

Comment (ergurjeet) [reply src/exchangeGroups.js:23]: > That is to allow if you only want to persist messages in 1 exchange, say only in `rejectedExchange`

~~Yes, my bad~~ That is like prefer exchange setting (enable/disable), or use group setting

**Review by vtovar — APPROVED** [review]
Looks good to me, but I dont know a lot about this functionality

**Review by normrider2001 — CHANGES_REQUESTED** [review]
You need to increment the version number on the package.json so you can deploy.

This is a minor change in reality, but to be on the safe side we should maybe make this major.

**Comment by normrider2001** [comment]
I think you should add some more information about this to the readme of this repo.

**Review by normrider2001 — APPROVED** [review]

**Pull request merged**